.circleci/continue_config.yml

```diff
@@ -1098,33 +1103,56 @@ jobs:
     # this job run pre-commit for Python client only
     # add dependency to other files when pre-commit runs related checks
     working_directory: ~/openlineage
-    docker:
-      - image: cimg/python:3.8
+    machine:
+      image: ubuntu-2404:current
     steps:
       - *checkout_project_root
       - restore_cache:
+          name: Restore pre-commit cache
+          keys:
+          - v1.4-precommit-deps-{{ checksum ".pre-commit-config.yaml" }}
+      - restore_cache:
+          name: Restore pmd cache
           keys:
-          - v1.2-precommit-deps-{{ checksum ".pre-commit-config.yaml" }}
+          - v1.0-pmd-{{ checksum ".pre_commit/run-pmd.sh" }}
+          - v1.0-pmd-
+      # - restore_cache:
+      #     name: Restore Gradle cache from Java client
+      #     keys:
+      #     - v1-client-java-
+      - restore_cache:
+          name: Restore Gradle cache from Spark integration
+          keys:
+          - v1-integration-spark-
```

Comment:
is this key correct?

Reply:
This will load most recent one. I don't think we care that much about specific version of gradle cache as it's mostly about getting dependencies loaded.